code/game/objects/items/weapons/cigs_lighters.dm

```diff
@@ -589,3 +589,28 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 			user << "<span class='warning'>You need to dry this first!</span>"
 	else
 		..()
+///////////////////
+//Strike anywhere//
+///////////////////
+/obj/item/clothing/mask/cigarette/strikeanywhere
+	name = "strike anywhere cigarette"
+	desc = "A cigarette infused with a proprietary compound that ignites when struck. Smells of magnesium."
+
+/obj/item/clothing/mask/cigarette/strikeanywhere/afterattack(atom/target, mob/user, proximity)
+	if(!lit && smoketime > 0)
+		user.visible_message("[user] strikes [target] with the [src] and it ignites!", "<span class='notice'>You strike [target] with the [src] and it ignites!</span>")
+		light()
+	..()
+
+/obj/item/clothing/mask/cigarette/strikeanywhere/throw_impact(atom/hit_atom)
+	if(!lit && smoketime > 0)
+		hit_atom.visible_message("The [src] strikes [hit_atom] and ignites!")
+		light()
+	..()

```

Comment:
@supermichael777
The trick to simulating multiple clients is to run the game in dream daemon, join the game signed into your byond account, then sign out of your account in the pager and join the game as a guest account without closing the original window with your signed in one. You'll get a popup yelling at you about multikeying, but that's it.
It's a super useful local debugging technique.